src/main/java/uk/ac/ebi/atlas/experimentpage/markergenes/MarkerGeneService.java

```diff
@@ -1,16 +1,40 @@
 package uk.ac.ebi.atlas.experimentpage.markergenes;
 
 import com.google.common.collect.ImmutableList;
+import org.springframework.stereotype.Service;
+import uk.ac.ebi.atlas.search.CellTypeSearchDao;
 
 import java.util.List;
 
-public interface MarkerGeneService {
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+@Service
+public class MarkerGeneService {
+    private final MarkerGenesDao markerGenesDao;
+    private final CellTypeSearchDao cellTypeSearchDao;
+
+    public MarkerGeneService(MarkerGenesDao markerGenesDao, CellTypeSearchDao cellTypeSearchDao) {
+        this.markerGenesDao = markerGenesDao;
+        this.cellTypeSearchDao = cellTypeSearchDao;
+    }
+
     /**
-     * Get marker genes for an experiment of that organism part.
+     * Get ImmutableList of marker genes objects for an experiment of that organism part.
      *
-     * @param experimentAccession
      * @param organismPart
+     * @param experimentAccession
      * @return ImmutableList of CellTypeMarkerGene Objects
      */
-    ImmutableList<CellTypeMarkerGene> getCellTypeMarkerGeneProfile(String experimentAccession, String organismPart);
+    public ImmutableList<CellTypeMarkerGene> getCellTypeMarkerGeneProfile(String experimentAccession, String organismPart) {
+        List<CellTypeMarkerGene> cellTypeMarkerGenes;
```

Comment:
This is just the same as initialising the variable to `null`. You don’t want that. If you need to (spoiler: you don’t), initialise it to an empty list which will be safe to use.

In fact, you can get rid of the variable declaration and the `if` block:
```java
var ontologyLabelsCellTypeValues = cellTypeSearchDao.getInferredCellTypeOntologyLabels(experimentAccession, organismPart);
        
var cellTypeMarkerGenes = ontologyLabelsCellTypeValues.isEmpty() ?
        markerGenesDao.getCellTypeMarkerGenes(experimentAccession, 
                 cellTypeSearchDao.getInferredCellTypeAuthorsLabels(experimentAccession, organismPart)) :
        markerGenesDao.getCellTypeMarkerGenes(experimentAccession, ontologyLabelsCellTypeValues);
                
return cellTypeMarkerGenes.stream()
        .filter(markerGene -> !markerGene.cellGroupValue().equalsIgnoreCase("Not available"))
        .collect(toImmutableList());
```

Use the ternary operator when you find a variable that takes a value depending on some condition, so that declaration and definition are together.

Reply:
Good catch :). I fixed it.